pkg/k8shandler/prometheus_rule.go

```diff
@@ -0,0 +1,102 @@
+package k8shandler
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/client/monitoring/v1"
+	v1alpha1 "github.com/openshift/elasticsearch-operator/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sYAML "k8s.io/apimachinery/pkg/util/yaml"
+)
+
+const (
+	alertsFilePath = "files/prometheus_alerts.yml"
+	rulesFilePath  = "files/prometheus_rules.yml"
+)
+
+func CreateOrUpdatePrometheusRules(dpl *v1alpha1.Elasticsearch) error {
+	ruleName := fmt.Sprintf("%s-%s", dpl.Name, "prometheus-rules")
+	owner := asOwner(dpl)
+
+	err := createOrUpdatePrometheusRules(ruleName, dpl.Namespace, dpl.Labels, owner)
+	if err != nil {
+		return fmt.Errorf("error creating PrometheusRule object: %v", err)
+	}
+
+	return nil
+}
+
+func createOrUpdatePrometheusRules(ruleName, namespace string, labels map[string]string, owner metav1.OwnerReference) error {
+	promRule, err := prometheusRule(ruleName, namespace, labels)
+	if err != nil {
+		return err
+	}
+
+	addOwnerRefToObject(promRule, owner)
+
+	err = sdk.Create(promRule)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+	//TODO: handle update
+	return nil
+}
+
+func prometheusRule(ruleName string, namespace string, labels map[string]string) (*monitoringv1.PrometheusRule, error) {
+	alertsRuleSpec, err := ruleSpec(alertsFilePath)
+	if err != nil {
+		return nil, err
+	}
+	rulesRuleSpec, err := ruleSpec(rulesFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	alertsRuleSpec.Groups = append(alertsRuleSpec.Groups, rulesRuleSpec.Groups...)
+
+	rule := monitoringv1.PrometheusRule{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       monitoringv1.PrometheusRuleKind,
+			APIVersion: monitoringv1.Group + "/" + monitoringv1.Version,
```

Comment:
`monitoringv1.SchemeGroupVersion.String()` instead

Reply:
Where did you find it?
`monitoringv1` only defines `Group` and `Version`

Reply:
https://github.com/openshift/elasticsearch-operator/blob/master/pkg/apis/elasticsearch/v1alpha1/register.go#L20